core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -877,11 +877,33 @@ private boolean checkConsistencyInternal(Inode inode, AlluxioURI path)
     }
     // TODO(calvin): Evaluate which other metadata fields should be validated.
     if (inode.isDirectory()) {
-      return ufs.isDirectory(ufsPath);
+      if (!ufs.isDirectory(ufsPath)) {
+        return false;
+      } else {
+        UfsFileStatus fileStatus = ufs.getFileStatus(ufsPath);
+        if (fileStatus.getMode() == inode.getMode()
```

Comment:
replace the if-else with `return fileStatus.getMode() == inode.getMode()&&fileStatus.getGroup().equals(inode.getGroup())&&fileStatus.getOwner().equals(inode.getOwner())`

Reply:
Use `Objects.equal` but not `String.equals` in case the group or owner is null